staker-rewards/test/rewards-calculator.test.js

```diff
@@ -164,4 +180,332 @@ describe("rewards calculator", async () => {
+    const mockContext = createMockContext()
+
+    setupContractsMock(mockContext)
+
+    const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+    operatorParameters.requirements.poolDeauthorizedInInterval = true
+
+    const rewardsCalculator = await RewardsCalculator.initialize(
+      mockContext,
+      interval,
+      [operatorParameters]
+    )
+
+    const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+    assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), true)
+  })
+
+  it("should set total rewards to zero if no minimum stake at start", async () => {
+    const mockContext = createMockContext()
+
+    setupContractsMock(mockContext)
+
+    const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+    operatorParameters.requirements.minimumStakeAtStart = false
+
+    const rewardsCalculator = await RewardsCalculator.initialize(
+      mockContext,
+      interval,
+      [operatorParameters]
+    )
+
+    const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+    assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), true)
+  })
+
+  it("should set total rewards to zero if pool requirement is not fulfilled at start", async () => {
+    const mockContext = createMockContext()
+
+    setupContractsMock(mockContext)
+
+    const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+    operatorParameters.requirements.poolRequirementFulfilledAtStart = false
+
+    const rewardsCalculator = await RewardsCalculator.initialize(
+      mockContext,
+      interval,
+      [operatorParameters]
+    )
+
+    const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+    assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), true)
+  })
+
+  it(
+    "should NOT set total rewards to zero if keygenCount >= 10 and " +
+      "failures percentage doesn't exceed 10%",
+    async () => {
+      const mockContext = createMockContext()
+
+      setupContractsMock(mockContext)
+
+      const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+      operatorParameters.operatorSLA.keygenCount = 10
+      operatorParameters.operatorSLA.keygenFailCount = 1
+      operatorParameters.operatorSLA.keygenSLA = 90
+
+      const rewardsCalculator = await RewardsCalculator.initialize(
+        mockContext,
+        interval,
+        [operatorParameters]
+      )
+
+      const rewards = rewardsCalculator.getOperatorRewards(operator)
+
+      assert.equal(rewards.totalRewards.isEqualTo(new BigNumber(0)), false)
+    }
+  )
+
+  it(
+    "should set total rewards to zero if keygenCount >= 10 and " +
+      "failures percentage exceeds 10%",
+    async () => {
+      const mockContext = createMockContext()
+
+      setupContractsMock(mockContext)
+
+      const operatorParameters = createOperatorParameters(operator, 70000, 100)
+
+      operatorParameters.operatorSLA.keygenCount = 10
```

Comment:
Can we have a separate test for `== 10` and `>= 10`?

The current format is a little confusing because test name says about `>=10` while we test only `== 10` boundary condition.

Reply:
Isn't it redundant? If we test for `11` we can test for `12` as well.

Reply:
Yeah, you are right.